automation/taskcluster/lib/tasks.py

```diff
@@ -353,21 +362,21 @@ def _craft_build_ish_task(
             treeherder=treeherder,
         )
 
-    def _craft_signing_task(self, name, description, signing_type, assemble_task_id, apk_paths, routes, treeherder):
+    def _craft_signing_task(self, name, description, signing_type, assemble_task_label, apk_paths, routes, treeherder):
         signing_format = "autograph_apk"
         payload = {
             'upstreamArtifacts': [{
                 'paths': apk_paths,
                 'formats': [signing_format],
-                'taskId': assemble_task_id,
+                'taskId': {'task-reference': '<build>'},
                 'taskType': 'build'
             }]
         }
 
         return self._craft_default_task_definition(
             worker_type='mobile-signing-dep-v1' if signing_type == 'dep' else 'mobile-signing-v1',
             provisioner_id='scriptworker-prov-v1',
-            dependencies=[assemble_task_id],
+            dependencies={'build': assemble_task_label},
```

Comment:
Can you do `task-reference` here?

Reply:
This a great question. I cannot do `task-reference` here, because that's actually the part which makes `task-reference` work. In other words, every time taskgraph sees `task-reference` somewhere in the task definition, it reads the task dependencies and looks what the alias is for. The dependencies dictionary follows this model `{alias: task_label}`. 

Here I defined the alias `build`, which allows me to use `'taskId': {'task-reference': '<build>'}` a few lines above.

For more information, here's the bits of code handling it: https://searchfox.org/mozilla-central/rev/d1e33e3e11f559952d7d80e722d26a6cf5dd80ac/taskcluster/taskgraph/util/parameterization.py#51